tasks/15.txt

```diff
@@ -1,3 +1,27 @@
 Create a SQL query to return the total number of orders for every month in 1997 year:
 
-| January | February | March | April | May | June | July | August | September | November | December |
\ No newline at end of file
+| January | February | March | April | May | June | July | August | September | November | December |
+
+db.orders.aggregate([
+    {
+        $match: { OrderDate: {$gte: ISODate("1997-01-01T00:00:00.000Z"), $lt: ISODate("1998-01-01T00:00:00.000Z") } }
+    },
+    {
+        $group: 
+        { 
+            _id: "",
+            "January": { "$sum": { "$cond": [ { "$eq": [ { $month: "$OrderDate" }, 1 ] }, 1, 0 ] } },
+            "February": { "$sum": { "$cond": [ { "$eq": [ { $month: "$OrderDate" }, 2 ] }, 1, 0 ] } },
+            "March": { "$sum": { "$cond": [ { "$eq": [ { $month: "$OrderDate" }, 3 ] }, 1, 0 ] } },
+            "April": { "$sum": { "$cond": [ { "$eq": [ { $month: "$OrderDate" }, 4 ] }, 1, 0 ] } },
+            "May": { "$sum": { "$cond": [ { "$eq": [ { $month: "$OrderDate" }, 5 ] }, 1, 0 ] } },
+            "June": { "$sum": { "$cond": [ { "$eq": [ { $month: "$OrderDate" }, 6 ] }, 1, 0 ] } },
+            "July": { "$sum": { "$cond": [ { "$eq": [ { $month: "$OrderDate" }, 7 ] }, 1, 0 ] } },
+            "August": { "$sum": { "$cond": [ { "$eq": [ { $month: "$OrderDate" }, 8 ] }, 1, 0 ] } },
+            "September": { "$sum": { "$cond": [ { "$eq": [ { $month: "$OrderDate" }, 9 ] }, 1, 0 ] } },
+            "October": { "$sum": { "$cond": [ { "$eq": [ { $month: "$OrderDate" }, 10 ] }, 1, 0 ] } },
+            "November": { "$sum": { "$cond": [ { "$eq": [ { $month: "$OrderDate" }, 11 ] }, 1, 0 ] } },
+            "December": { "$sum": { "$cond": [ { "$eq": [ { $month: "$OrderDate" }, 12 ] }, 1, 0 ] } }, 
+        }
+    }
+])
```

Comment:
Fine for now, but in real case will be better just group by { $month: "$OrderDate" } and then on server side map months numbers to names. You are recalculating/rechecking all records for each of fields now.